### PR TITLE
Fix bug preventing plans from downloading

### LINKED ISSUE #1844: Downloading my own plans produces empty files
For all file formats I get an empty file with just the Project Title when I try to download a plan. The project details coversheet info is populated correctly if I select that box (screenshot) but all answers and other content are missing. https://roadmap-stg.cdlib.org/plans/12849/download

screen shot 2018-08-24 at 2 29 00 pm

### PR DESCRIPTION
**Pull request by Bodacious**
Fixes #1844 .

Changes proposed in this PR:
- Fix bug preventing plans from downloading
- Refactor plan downloading to its own controller

### CODE REVIEW

File: app/models/phase.rb [hunk 1]
```diff
@@ -98,8 +98,7 @@ def deep_copy(**options)
   # TODO: Move this to Plan model as `num_answered_questions(phase=nil)`
   # Returns the number of answered question for the phase.
   def num_answered_questions(plan)
-    return 0 if plan.nil?
-    sections.to_a.sum { |s| s.num_answered_questions(plan) }
+    plan&.num_answered_questions.to_i
```
Comment (briri) [inline app/models/phase.rb]: You just changed my life :)

I grew very attached to this safe nav approach when working with Groovy years ago `obj?.method`. Glad to see its now a part of Ruby!

File: app/views/paginable/plans/_organisationally_or_publicly_visible.html.erb [hunk 2]
```diff
@@ -1,33 +1,45 @@
...
+                <%= paginable_sort_link('plans.updated_at') %>
+              </th>
+              <th scope="col" class="sorter-false"><%= _('Download') %></th>
+            </tr>
+          </thead>
+          <tbody>
+            <% scope.each do |plan| %>
+              <tr>
+                <td>
+                  <%= truncate plan.title, length: 40 %>
```
Comment (briri) [inline app/views/paginable/plans/_organisationally_or_publicly_visible.html.erb]: 👍 Much cleaner

File: app/views/paginable/plans/_privately_visible.html.erb [hunk 3]
```diff
@@ -16,7 +16,7 @@
           </thead>
           <tbody>
             <% scope.each do |plan| %>
-              <tr>
+              <tr id="<%= dom_id(plan) %>">
                 <td>
                   <%= link_to "#{plan.title.length > 60 ? "#{plan.title[0..59]} ..." : plan.title}",
```
Comment (briri) [inline app/views/paginable/plans/_privately_visible.html.erb]: Looks like another spot that would benefit from `truncate`

File: app/views/paginable/plans/_privately_visible.html.erb [hunk 4]
```diff
@@ -16,7 +16,7 @@
           </thead>
           <tbody>
             <% scope.each do |plan| %>
-              <tr>
+              <tr id="<%= dom_id(plan) %>">
```
Comment (briri) [inline app/views/paginable/plans/_privately_visible.html.erb]: It causes no harm, just curious why you would introduce an id attribute here. I don't see any JS that refers to it.

Comment (Bodacious) [reply app/views/paginable/plans/_privately_visible.html.erb]: It's used in the specs to make sure we click the correct link.

File: app/models/phase.rb (hunk 1 above)
Comment (Bodacious) [reply app/models/phase.rb]: Yep. It's not the prettiest syntax, but a very useful tool!

**Pull request merged**